workflows/differentialabundance.nf

```diff
@@ -454,6 +466,38 @@ workflow DIFFERENTIALABUNDANCE {
             .mix(GSEA_GSEA.out.versions)
     }
 
+    if (params.gprofiler2_run) {
+        ch_org_source = Channel.value([ params.gprofiler2_organism, params.gprofiler2_sources ])
+        if (!params.gprofiler2_background_file) {
+            // If param not set, use empty list as "background"
+            ch_background = []
+        } else if (params.gprofiler2_background_file == "auto") {
+            // If auto, use input matrix as background
+            if(params.study_type == "geo_soft_file") {
+                ch_background = ch_norm.map{it.tail()}
+            } else {
+                ch_background = ch_raw.map{it.tail()}
+            }
+        } else {
+            if (!file(params.gprofiler2_background_file).exists()) {
+                error("Param gprofiler2_background_file must be null, 'auto' or valid file path!")
+            }
+            ch_background = Channel.fromPath(params.gprofiler2_background_file)
+        }
+        if (!params.gprofiler2_gmt_file) {
```

Comment:
There is already a param `gsea_gene_sets` for providing GMT files to the pipeline. Let's just make that more generic so you can use it (and the `ch_gene_sets` channel) for gprofiler too.